velox/vector/BaseVector.h

```diff
@@ -177,6 +177,10 @@ class BaseVector {
     return type_;
   }
 
+  void setType(TypePtr type) {
```

Comment:
It would be sensible to add such an API, but it needs to happen in a separate PR with a test and with sanity checks. This API should check that the new type is not null and is compatible with the physical layout of the vector, e.g. type_->kindEquals(type).

It is Ok to change INTEGER() to DATE(), but not Ok to change VARCHAR() to REAL() for example.

Reply:
@mbasmanova https://github.com/facebookincubator/velox/pull/7783 add the setType() API in BaseVector.